### PR TITLE
Store currently logged in user in redux store

### PR DESCRIPTION
**Pull request by dustinnewman98**
## Description
Makes an API request to `8080current/user/` to retrieve the information on the current user and dispatch that information to the Redux store.

Resolves issue #47

### CODE REVIEW

File: client/src/actions/actionCreators.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ export function submitTicket(ticket) {
  * @description Calls the API method to get username and update store
  */
 export function getTest(cb) {
+	console.log('get test called');
```
Comment (TimothyGu) [inline client/src/actions/actionCreators.js]: \~clears throat\~

File: docker-compose.yml [hunk 2]
```diff
@@ -12,9 +12,9 @@ services:
...
-      - "3000:3000"
-    links:
-      - api
\ No newline at end of file
+  # web_client:
+  #   build: ./client
+  #   ports:
+  #     - "3000:3000"
+  #   links:
+  #     - api
```
Comment (TimothyGu) [inline docker-compose.yml]: ahem

File: client/src/reducers/user.js [hunk 3]
```diff
@@ -5,6 +5,10 @@ export default function UserReducer(state = { mentors: [] }, action) {
 			...state,
 			test: action.test
 		};
+	case 'GET_CURRENT_USER':
+		return {
+			...state,
+			current: action.user
```
Comment (TimothyGu) [inline client/src/reducers/user.js]: This is not valid JavaScript. (Missing })

File: client/src/actions/actionCreators.js [hunk 4]
```diff
@@ -23,9 +23,16 @@ export function getTest(cb) {
...
 
-/**
- * @description Pings the socket method socket/test
- */
+
+export function getCurrentUser() {
+	return (dispatch, prevState) => { // eslint-disable-line
+		api.getCurrentUser()
+			.then(response => {
+				dispatch({ type: 'GET_CURRENT_USER', current: response });
```
Comment (shannonphu) [inline client/src/actions/actionCreators.js]: Maybe rename `current` to `currentUser`. Also maybe the reducer here is trying to set who the current user is. Maybe rename the reducer to `SET_CURRENT_USER`. This **action** fetches the user, but the reducer stores the user into the store.

File: client/src/actions/createTicket.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export const createTicket = (Ticket) => {
```
Comment (shannonphu) [inline client/src/actions/createTicket.js]: Why is there a separate action creator here? Also creating a new ticket has nothing to do with this PR.

File: client/src/components/App.js [hunk 6]
```diff
@@ -16,6 +16,7 @@ function mapStateToProps(state) {
 }
 
 function mapDispatchToProps(dispatch) {
+	dispatch(actionCreators.getTest());
```
Comment (shannonphu) [inline client/src/components/App.js]: Remove this test call.

File: client/src/reducers/user.js [hunk 7]
```diff
@@ -1,10 +1,19 @@
 export default function UserReducer(state = { mentors: [] }, action) {
 	switch (action.type) {
 	case 'USER_TEST':
-		return {
+		return [
```
Comment (shannonphu) [inline client/src/reducers/user.js]: For USER_TEST, you don't have to change it to an array

File: client/src/reducers/user.js [hunk 8]
```diff
@@ -1,10 +1,19 @@
 export default function UserReducer(state = { mentors: [] }, action) {
```
Comment (shannonphu) [inline client/src/reducers/user.js]: Let's make the initial state more explicit by adding `current: null`

File: client/src/reducers/user.js [hunk 9]
```diff
@@ -1,10 +1,19 @@
...
+		return [
 			...state,
-			test: action.test
-		};
+			{
+				test: action.test
+			}
+		];
+	case 'GET_CURRENT_USER':
+		return [
```
Comment (shannonphu) [inline client/src/reducers/user.js]: I think you can actually do
```
return {
    ...state,
    current: action.user
}
```

**Comment by shannonphu** [comment]
Closed due to Git history issues

**Pull request closed without merging**